libraries/AP_BattMonitor/AP_BattMonitor_Logging.cpp

```diff
@@ -0,0 +1,70 @@
+#include "AP_BattMonitor_Backend.h"
+#include <AP_Logger/AP_Logger.h>
+
+extern const AP_HAL::HAL& hal;
+
+// Write BAT data packet(s)
+void AP_BattMonitor_Backend::Log_Write_BAT(const uint8_t instance, const uint64_t time_us) const
+{
+    bool has_temp = (AP_HAL::millis() - _state.temperature_time) <= AP_BATT_MONITOR_TIMEOUT;
```

Comment:
```suggestion
    float temp;
```
With the comment about using the helper below. If you find it particularly more readable you can extract it here, but we shouldn't be using the time calculation ourself and rely upon the helper to do it. (Particularly because has_temp should actually move to being a per backend call in the future, which would allow instances to be more clever/correct about what's reported here.)

Reply:
Added has_temp method to backend and used that instead